### PR TITLE
Call the new trento authentication script

### PR DESCRIPTION
**Pull request by hadeskun**
Starting using the new authentication script on trento.pm file, change the name of the script and some arguments.

- Related ticket: CFSA-1726
- Verification run: here the script called https://openqaworker15.qa.suse.cz/tests/131816#step/install_agents/30 and here the output https://openqaworker15.qa.suse.cz/tests/131816#step/install_agents/34 and here the key used from the output in the next command https://openqaworker15.qa.suse.cz/tests/131816#step/install_agents/55

### CODE REVIEW

File: lib/trento.pm [hunk 1]
```diff
@@ -768,11 +768,12 @@ Get the api-key from the Trento installation
 
 sub trento_api_key {
     my ($wd, $basedir) = @_;
-    my $cmd = join(' ', $basedir . '/trento-server-api-key.sh',
+    my $cmd = join(' ', $basedir . 'trento_deploy/trento_deploy.py',
```
Comment (alvarocarvajald) [inline lib/trento.pm:771]: I see in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/sles4sap/trento/install_agents.pm#L19-L23 that this is called with `$basedir = '/root/test'`, so unless we expect a directory called `/root/testtrento_deploy/...` to exist, then the line above is missing a `/` right after `$basedir`.

I was going to check how that's working in the VRs, but they have not been added yet.

File: lib/trento.pm [hunk 2]
```diff
@@ -768,11 +768,12 @@ Get the api-key from the Trento installation
...
+        '--verbose', 'api_key',
         '-u', 'admin',
         '-p', get_trento_password(),
         '-i', get_trento_ip(),
-        '-d', $wd, '-v');
```
Comment (alvarocarvajald) [inline lib/trento.pm:775]: If `$wd` is no longer needed, shouldn't it be removed as an argument?

File: lib/trento.pm (hunk 1 above)
Comment (hadeskun) [reply lib/trento.pm]: Good obervation master thx, refactored

File: lib/trento.pm (hunk 2 above)
Comment (hadeskun) [reply lib/trento.pm:775]: Nice comment thx master :)

File: lib/trento.pm [hunk 3]
```diff
@@ -767,12 +767,13 @@ Get the api-key from the Trento installation
...
-    my ($wd, $basedir) = @_;
-    my $cmd = join(' ', $basedir . '/trento-server-api-key.sh',
+    my ($basedir) = @_;
+    my $cmd = join(' ', $basedir . '/trento_deploy/trento_deploy.py',
+        '--verbose', 'api_key',
         '-u', 'admin',
         '-p', get_trento_password(),
         '-i', get_trento_ip(),
-        '-d', $wd, '-v');
+        '-v');
```
Comment (mpagot) [inline lib/trento.pm:776]: I think that this last `-v` is not needed too

File: lib/trento.pm [hunk 4]
```diff
@@ -767,12 +767,13 @@ Get the api-key from the Trento installation
 =cut
 
 sub trento_api_key {
-    my ($wd, $basedir) = @_;
-    my $cmd = join(' ', $basedir . '/trento-server-api-key.sh',
+    my ($basedir) = @_;
```
Comment (mpagot) [inline lib/trento.pm:770]: if you remove `$wd` from here (that is fine) you also need to remove it from the caller https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/25bec175868356be70cde4d3796c2a02fd4f4256/tests/sles4sap/trento/install_agents.pm#L23

**Comment by mpagot** [comment]
You can try to add a unit testing function in *t/08_trento.pm* to help you in changing the code. As reference starting point a test that is passing with the code before your PR

```
subtest '[trento_api_key]' => sub {
    my $trento = Test::MockModule->new('trento', no_auto => 1);
    @calls = ();

    $trento->redefine(script_output => sub {
        push @calls, $_[0];
        return 'GARBAGE
GARBAGE
api_key:SUPERSECRETINGREDIENT';});
    $trento->redefine(get_trento_password => sub { return 'PIZZOCCHERI'; });
    $trento->redefine(get_trento_ip => sub { return 'BITTO'; });

    my $key = trento_api_key('/PEPERONATA', '/RATATUILLE');

    note("\n  C-->  " . join("\n  C-->  ", @calls));

    like $calls[0], qr/\/RATATUILLE\/trento-server-api-key.sh/;
    like $calls[0], qr/\-u admin/;
    like $calls[0], qr/\-p PIZZOCCHERI/;
    like $calls[0], qr/\-i BITTO/;
    is $key, "SUPERSECRETINGREDIENT";
};
```

File: lib/trento.pm (hunk 4 above)
Comment (alvarocarvajald) [reply lib/trento.pm:770]: It should still work, but I agree that it's better to remove it from the caller as well.

File: lib/trento.pm (hunk 3 above)
Comment (hadeskun) [reply lib/trento.pm:776]: thx for the comment master, will remove the argument.

**Review by mpagot — APPROVED** [review]
LGTM

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Pull request merged**